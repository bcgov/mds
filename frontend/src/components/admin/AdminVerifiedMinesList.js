import React from "react";
import { Table } from "antd";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import CustomPropTypes from "@/customPropTypes";
import * as router from "@/constants/routes";
import NullScreen from "@/components/common/NullScreen";
import { formatDate } from "@/utils/helpers";

/**
 * @class AdminVerifiedMinesList displays list of mineVerifiedStatuses for the admin page.
 */

const propTypes = {
  minesVerifiedStatusList: PropTypes.arrayOf(CustomPropTypes.mineVerificationStatus).isRequired,
};

const columns = [
  {
    title: "Mine Name",
    width: 300,
    dataIndex: "mine_name",
    render: (text, record) => (
      <div key={record.key}>
        <Link to={router.MINE_SUMMARY.dynamicRoute(record.key)}>{text}</Link>
      </div>
    ),
  },
  {
    title: "Last Verified By",
    width: 150,
    dataIndex: "verifying_user",
  },
  {
    title: "Last Verified On",
    width: 150,
    dataIndex: "formatted_timestamp",
  },
];

const transformRowData = (verifiedMinesList) =>
  verifiedMinesList.map(({ mine_guid, verifying_timestamp, ...rest }) => ({
    key: mine_guid,
    formatted_timestamp: formatDate(verifying_timestamp),
    ...rest,
  }));

export const AdminVerifiedMinesList = (props) => (
  <div>
    <Table
      align="center"
      pagination={false}
      columns={columns}
      dataSource={transformRowData(props.minesVerifiedStatusList)}
      scroll={{ y: 500 }}
      locale={{ emptyText: <NullScreen type="no-results" /> }}
    />
  </div>
);

AdminVerifiedMinesList.propTypes = propTypes;

export default AdminVerifiedMinesList;
