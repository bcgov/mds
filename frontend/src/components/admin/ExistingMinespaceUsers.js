import React from "react";
/**
 * @class AdminDashboard houses everything related to admin tasks, this is a permission-based route.
 */

export const ExistingMinespaceUsers = () => (
  <div>
    <h3>Minespace Users</h3>
  </div>
);

export default ExistingMinespaceUsers;
