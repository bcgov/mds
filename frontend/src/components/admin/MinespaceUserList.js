import React from "react";
import { Table, Button, Popconfirm, Icon } from "antd";
import PropTypes from "prop-types";
import CustomPropTypes from "@/customPropTypes";
import * as Strings from "@/constants/strings";
import NullScreen from "@/components/common/NullScreen";

const propTypes = {
  minespaceUsers: PropTypes.arrayOf(CustomPropTypes.minespaceUser),
  minespaceUserMines: PropTypes.arrayOf(CustomPropTypes.mineName),
  handleDelete: PropTypes.func,
};

const defaultProps = {
  minespaceUsers: [],
  minespaceUserMines: [],
  handleDelete: () => {},
};

const columns = [
  {
    title: "User Email",
    width: 400,
    dataIndex: "email",
  },
  {
    title: "Mines",
    width: 300,
    dataIndex: "mineNames",
    render: (text) => (
      <div>{text && text.map(({ guid, mine_name }) => <div key={guid}>{mine_name}</div>)}</div>
    ),
  },
  {
    title: "",
    width: 50,
    dataIndex: "delete",
    render: (text, record) => (
      <Popconfirm
        placement="topLeft"
        title={`Are you sure you want to delete ${record.email}?`}
        onConfirm={() => text(record.user_id)}
        okText="Delete"
        cancelText="Cancel"
      >
        <Button className="full-mobile" ghost type="primary">
          <Icon type="minus-circle" theme="outlined" />
        </Button>
      </Popconfirm>
    ),
  },
];

const lookupMineName = (mine_guids, mines) =>
  mine_guids.map((guid) => {
    const mine_record = mines.find((mine) => mine.guid === guid);
    return {
      guid,
      mine_name: mine_record ? `${mine_record.mine_name}-${mine_record.mine_no}` : "",
    };
  });

const transformRowData = (minespaceUsers, mines, deleteFunc) =>
  minespaceUsers.map((user) => ({
    key: user.user_id,
    emptyField: Strings.EMPTY_FIELD,
    email: user.email,
    mineNames: lookupMineName(user.mines, mines),
    user_id: user.user_id,
    delete: deleteFunc,
  }));

export const MinespaceUserList = (props) => (
  <div>
    {props.minespaceUserMines && (
      <Table
        align="center"
        pagination={false}
        columns={columns}
        dataSource={transformRowData(
          props.minespaceUsers,
          props.minespaceUserMines,
          props.handleDelete
        )}
        scroll={{ x: 1500 }}
        locale={{ emptyText: <NullScreen type="no-results" /> }}
      />
    )}
  </div>
);
MinespaceUserList.propTypes = propTypes;
MinespaceUserList.defaultProps = defaultProps;

export default MinespaceUserList;
