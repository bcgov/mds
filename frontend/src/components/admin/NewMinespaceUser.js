import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import AddMinespaceUser from "@/components/Forms/AddMinespaceUser";
import CustomPropTypes from "@/customPropTypes";
import { fetchMineNameList } from "@/actionCreators/mineActionCreator";
import { getMineNames } from "@/selectors/mineSelectors";
import { createMinespaceUser } from "@/actionCreators/minespaceActionCreator";

const propTypes = {
  fetchMineNameList: PropTypes.func.isRequired,
  mines: PropTypes.arrayOf(CustomPropTypes.mineName),
  createMinespaceUser: PropTypes.func.isRequired,
  refreshData: PropTypes.func,
};

const defaultProps = {
  mines: [],
  refreshData: () => {},
};

export class NewMinespaceUser extends Component {
  componentDidMount() {
    this.props.fetchMineNameList();
  }

  createNewBCEIDUser = (values) => {
    const payload = {
      mine_guids: values.proponent_mine_access.map((val) => val.split("~")[1]),
      email: values.user_bceid_email,
    };

    this.props.createMinespaceUser(payload).then(() => {
      this.props.refreshData();
    });
  };

  handleSearch = (name) => {
    if (name.length > 0) {
      this.props.fetchMineNameList({ name });
    }
  };

  handleChange = () => {
    this.props.fetchMineNameList();
  };

  render() {
    return (
      <div>
        <h3>Add BCeID User</h3>
        {this.props.mines && (
          <AddMinespaceUser
            mines={this.props.mines.map((mine) => ({
              value: `${mine.mine_name}~${mine.mine_guid}`,
              label: `${mine.mine_name}-${mine.mine_no}`,
            }))}
            onSubmit={this.createNewBCEIDUser}
            handleChange={this.handleChange}
            handleSearch={this.handleSearch}
          />
        )}
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  mines: getMineNames(state).mines,
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchMineNameList,
      createMinespaceUser,
    },
    dispatch
  );

NewMinespaceUser.propTypes = propTypes;
NewMinespaceUser.defaultProps = defaultProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NewMinespaceUser);
