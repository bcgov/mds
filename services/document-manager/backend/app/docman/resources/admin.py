import numpy
import uuid

from werkzeug.exceptions import Forbidden
from flask import current_app
from flask_restplus import Resource, reqparse
from celery import chord

from app.docman.models.document import Document
from app.utils.access_decorators import requires_any_of, MINE_ADMIN
from app.services.object_store_storage_service import ObjectStoreStorageService
from app.config import Config


class TransferDocsToObjectStore(Resource):
    parser = reqparse.RequestParser(trim=True)
    parser.add_argument('secret', type=str, required=True)

    @requires_any_of([MINE_ADMIN])
    def post(self):
        # Ensure that the admin API secret required to initiate the transfer is correct
        data = self.parser.parse_args()
        secret = data.get('secret')
        if (Config.ADMIN_API_SECRET is None or secret != Config.ADMIN_API_SECRET):
            raise Forbidden()

        return self.transfer_task()

    def transfer_task(self):
        from app.utils.tasks import transfer_docs, transfer_docs_result

        # Get the documents that aren't stored on the object store (return if they all are)
        docs = Document.query.filter_by(object_store_path=None).all()
        if len(docs) == 0:
            return 'No documents need to be transferred', 200

        # Split the list of documents to transfer into N chunks to upload in parallel
        chunks = numpy.array_split(docs, 8)
        chunks = [x.tolist() for x in chunks if len(x) > 0]

        # Create the transfer tasks
        tasks = []
        transfer_id = str(uuid.uuid4())
        for i, chunk in enumerate(chunks):
            doc_ids = [doc.document_id for doc in chunk]
            tasks.append(transfer_docs.subtask((transfer_id, doc_ids, i)))

        # Start the transfer tasks
        callback = transfer_docs_result.subtask(kwargs={'transfer_id': transfer_id})
        chord(tasks)(callback)

        # Create the response message
        message = f'Added transfer job with ID {transfer_id} to the task queue: {len(docs)} docs will be transferred in {len(chunks)} chunks of size {len(chunks[0])}'
        current_app.logger.info(message)
        current_app.logger.debug(chunks)

        return message, 202


class CompareDocsOnObjectStore(Resource):
    parser = reqparse.RequestParser(trim=True)
    parser.add_argument('secret', type=str, required=True)

    @requires_any_of([MINE_ADMIN])
    def post(self):
        # Ensure that the admin API secret required to initiate the verification is correct
        data = self.parser.parse_args()
        secret = data.get('secret')
        if (Config.ADMIN_API_SECRET is None or secret != Config.ADMIN_API_SECRET):
            raise Forbidden()

        return self.verify_task()

    def verify_task(self):
        from app.utils.tasks import verify_docs, verify_docs_result

        # Get the documents that are stored on the object store (return if there are none)
        docs = Document.query.filter(Document.object_store_path != None).all()
        if len(docs) == 0:
            return 'No documents are stored on the object store', 200

        # Split the list of documents to verify into N chunks to verify in parallel
        chunks = numpy.array_split(docs, 8)
        chunks = [x.tolist() for x in chunks if len(x) > 0]

        # Create the verification tasks
        tasks = []
        verify_id = str(uuid.uuid4())
        for i, chunk in enumerate(chunks):
            doc_ids = [doc.document_id for doc in chunk]
            tasks.append(verify_docs.subtask((verify_id, doc_ids, i)))

        # Start the verification tasks
        callback = verify_docs_result.subtask(kwargs={'verify_id': verify_id})
        chord(tasks)(callback)

        # Create the response message
        message = f'Added verification job with ID {verify_id} to the task queue: {len(docs)} docs will be verified in {len(chunks)} chunks of size {len(chunks[0])}'
        current_app.logger.info(message)
        current_app.logger.debug(chunks)

        return message, 202


class GetUntransferredFiles(Resource):
    parser = reqparse.RequestParser(trim=True)
    parser.add_argument('secret', type=str, required=True)

    @requires_any_of([MINE_ADMIN])
    def post(self):

        # Ensure that the admin API secret is correct
        data = self.parser.parse_args()
        secret = data.get('secret')
        if (Config.ADMIN_API_SECRET is None or secret != Config.ADMIN_API_SECRET):
            raise Forbidden()

        return self.get_untransferred_files()

    def get_untransferred_files(self):
        # Get and return the documents that aren't stored on the object store
        docs = Document.query.filter_by(object_store_path=None).all()
        doc_jsons = [doc.json() for doc in docs]
        return doc_jsons, 200
