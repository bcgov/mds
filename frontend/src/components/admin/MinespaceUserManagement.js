import React, { Component } from "react";
import { flatMap, uniq } from "lodash";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import MinespaceUserList from "@/components/admin/MinespaceUserList";
import NewMinespaceUser from "@/components/admin/NewMinespaceUser";
import CustomPropTypes from "@/customPropTypes";
import { getMineNames } from "@/selectors/mineSelectors";
import { getMinespaceUsers } from "@/selectors/minespaceSelector";
import { fetchMineNameList } from "@/actionCreators/mineActionCreator";
import {
  fetchMinespaceUsers,
  deleteMinespaceUser,
  fetchMinespaceUserMines,
} from "@/actionCreators/minespaceActionCreator";
import { getMinespaceUserMines } from "../../reducers/minespaceReducer";

/**
 * @class AdminDashboard houses everything related to admin tasks, this is a permission-based route.
 */
const propTypes = {
  minespaceUsers: PropTypes.arrayOf(CustomPropTypes.minespaceUser),
  minespaceUserMines: PropTypes.arrayOf(CustomPropTypes.mineName),
  fetchMineNameList: PropTypes.func.isRequired,
  fetchMinespaceUsers: PropTypes.func.isRequired,
  fetchMinespaceUserMines: PropTypes.func.isRequired,
  deleteMinespaceUser: PropTypes.func.isRequired,
};

const defaultProps = {
  minespaceUsers: [],
  minespaceUserMines: [],
};

export class MinespaceUserManagement extends Component {
  componentDidMount() {
    this.props.fetchMineNameList();
    this.refreshUserData();
  }

  handleDelete = (userId) => {
    this.props.deleteMinespaceUser(userId).then(() => {
      this.props.fetchMinespaceUsers();
    });
  };

  refreshUserData = () => {
    this.props.fetchMinespaceUsers().then(() => {
      const mine_guids = flatMap(this.props.minespaceUsers, (user) => user.mines);
      this.props.fetchMinespaceUserMines(uniq(mine_guids));
    });
  };

  render() {
    return (
      <div>
        <h2>Minespace User Management</h2>
        <NewMinespaceUser refreshData={this.refreshUserData} />
        <h3>Minespace Users</h3>
        <MinespaceUserList
          minespaceUsers={this.props.minespaceUsers}
          minespaceUserMines={this.props.minespaceUserMines}
          handleDelete={this.handleDelete}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  mines: getMineNames(state).mines,
  minespaceUsers: getMinespaceUsers(state),
  minespaceUserMines: getMinespaceUserMines(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchMineNameList,
      fetchMinespaceUsers,
      fetchMinespaceUserMines,
      deleteMinespaceUser,
    },
    dispatch
  );

MinespaceUserManagement.propTypes = propTypes;
MinespaceUserManagement.defaultProps = defaultProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MinespaceUserManagement);
