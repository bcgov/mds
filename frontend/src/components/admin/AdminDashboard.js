import React, { Component } from "react";
import { Input, Button } from "antd";
import { AuthorizationGuard } from "@/HOC/AuthorizationGuard";
import * as Permission from "@/constants/permissions";
import MinespaceUserManagement from "@/components/admin/MinespaceUserManagement";
import { downloadMineManagerHistory } from "@/actionCreators/partiesActionCreator";

/**
 * @class AdminDashboard houses everything related to admin tasks, this is a permission-based route.
 */

export class AdminDashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      mineNo: "",
    };
  }

  handleChange = (e) => {
    this.setState({
      mineNo: e.target.value,
    });
  };

  handleDownload = () => {
    downloadMineManagerHistory(this.state.mineNo, { window, document });
  };

  render() {
    return (
      <div className="landing-page">
        <div className="landing-page__header">
          <h1>Admin View</h1>
        </div>
        <div className="landing-page__content">
          <div className="tab__content">
            <h2>Pull Mine Manager History</h2>
            <Input
              placeholder="Enter a Mine Number"
              value={this.state.mineNo}
              onChange={this.handleChange}
              style={{ width: "50%", "max-width": "300px" }}
            />
            <Button onClick={this.handleDownload}>Download History</Button>
          </div>
          <div className="tab__content">
            <MinespaceUserManagement />
          </div>
        </div>
      </div>
    );
  }
}

export default AuthorizationGuard(Permission.ADMIN)(AdminDashboard);
