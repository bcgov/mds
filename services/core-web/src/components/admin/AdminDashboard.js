import React, { Component } from "react";
import { Row, Col, Tabs } from "antd";
import { compose, bindActionCreators } from "redux";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { fetchMineVerifiedStatuses } from "@common/actionCreators/mineActionCreator";
import { AuthorizationGuard } from "@/HOC/AuthorizationGuard";
import * as Permission from "@/constants/permissions";
import MinespaceUserManagement from "@/components/admin/MinespaceUserManagement";
import { AdminVerifiedMinesList } from "@/components/admin/AdminVerifiedMinesList";

/**
 * @class AdminDashboard houses everything related to admin tasks, this is a permission-based route.
 */

const { TabPane } = Tabs;

const propTypes = {
  fetchMineVerifiedStatuses: PropTypes.func.isRequired,
};

export class AdminDashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeTab: "verifiedMines",
      isLoaded: false,
      verifiedMines: [],
      unverifiedMines: [],
    };
  }

  componentWillMount() {
    this.props.fetchMineVerifiedStatuses().then((response) => {
      this.setState({
        isLoaded: true,
        verifiedMines: response.data
          .filter((vm) => vm.healthy_ind === true)
          .sort(this.compareMineName),
        unverifiedMines: response.data
          .filter((vm) => vm.healthy_ind === false)
          .sort(this.compareMineName),
      });
    });
  }

  handleTabChange = (activeTab) => {
    this.setState({
      activeTab,
    });
  };

  compareMineName = (a, b) => a.mine_name.localeCompare(b.mine_name);

  render() {
    return (
      <div className="landing-page">
        <div className="landing-page__header">
          <Row>
            <Col sm={22} md={14} lg={12}>
              <h1>Admin View</h1>
            </Col>
          </Row>
        </div>
        <div className="landing-page__content">
          <Tabs
            className="center-tabs"
            activeKey={this.state.activeTab}
            defaultActiveKey="summary"
            onChange={this.handleTabChange}
            size="large"
            animated={{ inkBar: true, tabPane: false }}
          >
            <TabPane tab="Verified Mines" key="verifiedMines">
              <div className="tab__content">
                <div>
                  <h4>{this.state.verifiedMines.length}&nbsp;Verified Mines</h4>
                  <div>
                    <AdminVerifiedMinesList
                      isLoaded={this.state.isLoaded}
                      minesVerifiedStatusList={this.state.verifiedMines}
                    />
                  </div>
                </div>
              </div>
            </TabPane>
            <TabPane tab="Unverified Mines" key="unverifiedMines">
              <div className="tab__content">
                <div>
                  <h4>{this.state.unverifiedMines.length}&nbsp;Mines Needing Re-Verification</h4>
                  <div>
                    <AdminVerifiedMinesList
                      isLoaded={this.state.isLoaded}
                      minesVerifiedStatusList={this.state.unverifiedMines}
                    />
                  </div>
                </div>
              </div>
            </TabPane>
            <TabPane tab="Manage MineSpace Users" key="managerMinespaceUsers">
              <div className="tab__content">
                <MinespaceUserManagement />
              </div>
            </TabPane>
          </Tabs>
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchMineVerifiedStatuses,
    },
    dispatch
  );

AdminDashboard.propTypes = propTypes;

export default compose(
  connect(null, mapDispatchToProps),
  AuthorizationGuard(Permission.ADMIN)
)(AdminDashboard);
