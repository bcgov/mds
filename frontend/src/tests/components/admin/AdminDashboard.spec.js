import React from "react";
import { shallow } from "enzyme";
import { AdminDashboard } from "@/components/admin/AdminDashboard";

describe("AdminDashboard", () => {
  it("renders properly", () => {
    const component = shallow(<AdminDashboard />);
    expect(component).toMatchSnapshot();
  });
});
