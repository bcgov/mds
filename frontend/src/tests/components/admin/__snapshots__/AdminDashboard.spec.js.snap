// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboard renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminDashboard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="landing-page__header"
        >
          <h1>
            Admin View
          </h1>
        </div>,
        <div
          className="landing-page__content"
        >
          <div
            className="tab__content"
          >
            <h2>
              Pull Mine Manager History
            </h2>
            <Input
              disabled={false}
              onChange={[Function]}
              placeholder="Enter a Mine Number"
              prefixCls="ant-input"
              style={
                Object {
                  "max-width": "300px",
                  "width": "50%",
                }
              }
              type="text"
              value=""
            />
            <Button
              block={false}
              ghost={false}
              loading={false}
              onClick={[Function]}
              prefixCls="ant-btn"
            >
              Download History
            </Button>
          </div>
          <div
            className="tab__content"
          >
            <Connect(MinespaceUserManagement) />
          </div>
        </div>,
      ],
      "className": "landing-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Admin View
          </h1>,
          "className": "landing-page__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Admin View",
          },
          "ref": null,
          "rendered": "Admin View",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="tab__content"
            >
              <h2>
                Pull Mine Manager History
              </h2>
              <Input
                disabled={false}
                onChange={[Function]}
                placeholder="Enter a Mine Number"
                prefixCls="ant-input"
                style={
                  Object {
                    "max-width": "300px",
                    "width": "50%",
                  }
                }
                type="text"
                value=""
              />
              <Button
                block={false}
                ghost={false}
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
              >
                Download History
              </Button>
            </div>,
            <div
              className="tab__content"
            >
              <Connect(MinespaceUserManagement) />
            </div>,
          ],
          "className": "landing-page__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Pull Mine Manager History
                </h2>,
                <Input
                  disabled={false}
                  onChange={[Function]}
                  placeholder="Enter a Mine Number"
                  prefixCls="ant-input"
                  style={
                    Object {
                      "max-width": "300px",
                      "width": "50%",
                    }
                  }
                  type="text"
                  value=""
                />,
                <Button
                  block={false}
                  ghost={false}
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                >
                  Download History
                </Button>,
              ],
              "className": "tab__content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Pull Mine Manager History",
                },
                "ref": null,
                "rendered": "Pull Mine Manager History",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "onChange": [Function],
                  "placeholder": "Enter a Mine Number",
                  "prefixCls": "ant-input",
                  "style": Object {
                    "max-width": "300px",
                    "width": "50%",
                  },
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "block": false,
                  "children": "Download History",
                  "ghost": false,
                  "loading": false,
                  "onClick": [Function],
                  "prefixCls": "ant-btn",
                },
                "ref": null,
                "rendered": "Download History",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(MinespaceUserManagement) />,
              "className": "tab__content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="landing-page__header"
          >
            <h1>
              Admin View
            </h1>
          </div>,
          <div
            className="landing-page__content"
          >
            <div
              className="tab__content"
            >
              <h2>
                Pull Mine Manager History
              </h2>
              <Input
                disabled={false}
                onChange={[Function]}
                placeholder="Enter a Mine Number"
                prefixCls="ant-input"
                style={
                  Object {
                    "max-width": "300px",
                    "width": "50%",
                  }
                }
                type="text"
                value=""
              />
              <Button
                block={false}
                ghost={false}
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
              >
                Download History
              </Button>
            </div>
            <div
              className="tab__content"
            >
              <Connect(MinespaceUserManagement) />
            </div>
          </div>,
        ],
        "className": "landing-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Admin View
            </h1>,
            "className": "landing-page__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Admin View",
            },
            "ref": null,
            "rendered": "Admin View",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="tab__content"
              >
                <h2>
                  Pull Mine Manager History
                </h2>
                <Input
                  disabled={false}
                  onChange={[Function]}
                  placeholder="Enter a Mine Number"
                  prefixCls="ant-input"
                  style={
                    Object {
                      "max-width": "300px",
                      "width": "50%",
                    }
                  }
                  type="text"
                  value=""
                />
                <Button
                  block={false}
                  ghost={false}
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                >
                  Download History
                </Button>
              </div>,
              <div
                className="tab__content"
              >
                <Connect(MinespaceUserManagement) />
              </div>,
            ],
            "className": "landing-page__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Pull Mine Manager History
                  </h2>,
                  <Input
                    disabled={false}
                    onChange={[Function]}
                    placeholder="Enter a Mine Number"
                    prefixCls="ant-input"
                    style={
                      Object {
                        "max-width": "300px",
                        "width": "50%",
                      }
                    }
                    type="text"
                    value=""
                  />,
                  <Button
                    block={false}
                    ghost={false}
                    loading={false}
                    onClick={[Function]}
                    prefixCls="ant-btn"
                  >
                    Download History
                  </Button>,
                ],
                "className": "tab__content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Pull Mine Manager History",
                  },
                  "ref": null,
                  "rendered": "Pull Mine Manager History",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "disabled": false,
                    "onChange": [Function],
                    "placeholder": "Enter a Mine Number",
                    "prefixCls": "ant-input",
                    "style": Object {
                      "max-width": "300px",
                      "width": "50%",
                    },
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "block": false,
                    "children": "Download History",
                    "ghost": false,
                    "loading": false,
                    "onClick": [Function],
                    "prefixCls": "ant-btn",
                  },
                  "ref": null,
                  "rendered": "Download History",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(MinespaceUserManagement) />,
                "className": "tab__content",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
